client/dashboard/app/queries/domain-dns-records.ts

```diff
@@ -1,5 +1,12 @@
-import { mutationOptions } from '@tanstack/react-query';
-import { updateDomainDns, DnsRecord } from '../../data/domain-dns-records';
+import { mutationOptions, queryOptions } from '@tanstack/react-query';
+import { updateDomainDns, DnsRecord, fetchDomainDns } from '../../data/domain-dns-records';
+import { queryClient } from '../query-client';
+
+export const domainDnsQuery = ( domainName: string ) =>
+	queryOptions( {
+		queryKey: [ 'domains', domainName, 'domain-dns' ],
```

Comment:
[nit] we can drop the `domain-` prefix since it's already the prefix 😄 


```suggestion
		queryKey: [ 'domains', domainName, 'dns' ],
```

Reply:
Good point! I rebased this PR after #105129 was merged and this was fixed 🙂